### ISSUE TITLE
Server Ops - documentation and DR checklists - let's build an index of ops docs!

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#2309]
An opportunity to review ops. For everything we run: lobby, bots, forum, github.io website, dice server, warclub website and forum; do we have:

### documentation:
- how to install
- how to check if running
- how to restart

### Disaster Recovery
- are we taking data backups?
- are data backups replicated to a second physical machine?

I think we can consider Derby dead for this discussion. For other legacy components like warclub mysql DB, either we should be able to recover that or continue with our plans to deprecate it post-haste. I'm a bit most interested to know if are backing up the forum data. We do not currently have something for postgres, this is a good reminder.

Beyond the conversation that results, which I think will be useful to get us on the same page, I hope we can create a master link index out of this maybe, so we can have all these sources of information readily available and organized.

**Comment by prastle** [comment]
The warclub and old forum have had 0 posts in 2 months other than some spam to the best of my knowledge.  ( Also off topic but could we get a couple bots in the test lobby so @Hepps can test trains in Global Dominance.  He is ready to test @DanVanAtta @RoiEXLab  )

**Comment by RoiEXLab** [comment]
The mongoDB backing our forum has no backup on it's own, but doesn't linode do regular backups on the servers?

**Comment by RoiEXLab** [comment]
Edit: I don't think so, but we could use their backup service for additional extra dollars per month

**Comment by RoiEXLab** [comment]
And well I don't think we need guides on the io website 😅
Github hosts this on it's own...

**Comment by DanVanAtta** [comment]
> And well I don't think we need guides on the io website :sweat_smile:
Github hosts this on it's own...

I think I see your point, though I disagree slightly. IMO we would not write a guide, but give a link to it. The idea being that nobody would have to hunt for this type of documentation. We did run into big trouble with the last few NodeBB crashes, nobody was able to get that up and going except one person. Not to pick on you @RoiEXLab , but this has been an ongoing problem in TripleA where one person will set something up, do the install, some configuration, and then we leave it running. In the case of warclub, nobody now knows fully how that was setup, and we are completely boned if we need to re-install it from scratch. That was pretty much the case for lobby and bots too, we have scripts there now though. Which is another part of the point of these guides, those scripts are not useful if nobody knows about them, can everyone beyond me use those scripts? I'm not 100% sure, indexing the script location though I think is a start.

**Comment by DanVanAtta** [comment]
> Edit: I don't think so, but we could use their backup service for additional extra dollars per month

It may make sense. Looks like we would want it for the forum and warclub servers, $4 / month. It's not a lot. I am a bit reluctant to solely rely on this though (maybe the carbon copy data backup is a good last resort to have for us):
- a custom solution would likely be more robust and would have no monthly cost
- we can practice and script a custom solution more easily I suspect. Any disaster recovery/backup mechanism needs to be tested regularly.

**Comment by RoiEXLab** [comment]
@DanVanAtta
> I think I see your point, though I disagree slightly. IMO we would not write a guide, but give a link to it.

Of course, what I meant was that on the io website we don't need to say something like "this is how you start, stop and update the server", but of course we should provide information that github.io is powered by jekyll, has limits which make dynamic handling of HTTP parameters impossible, and the only thing we need to do in order to keep it running is to not mess up the config file (which is being validated during the travis Build)
>  can everyone beyond me use those scripts?

Well I can't speak for everyone, but at least for me those scripts were straight forward to use thanks to your documentation in the README file.
But I must admit, that it would have been a lot harder to modify the lobby ip, if I didn't write the systemd script on my own, that's why I opened an issue about this...
Also I felt some of the scripts were in unintuitive folders: I expected the start_all script to be in the bots folder, because it's not saying what it's actually starting, and the correct folder would have given slightly more context.

TL;DR: I get your point, and once I have a little bit more time again, which should be in the next week, I'd be happy to share some of my knowledge about the forum system (and a little bit of xoops, I had to deal with it during the lobby server upgrade to Ubuntu 16.04), in the form of some documentation.

**Comment by DanVanAtta** [comment]
adding docs, reviewing, cleaning them, it's a continuous process as software changes - so we should always be revisiting this topic when we get teh chance : )

For now, there are no clear calls to action in the issue, it's been some time since last activity - closing.
